pipeline/parameters.py

```diff
@@ -121,6 +123,31 @@ def __init__(self, **kwargs):
             "verbose", 0, int, "Level of verbosity (0=quiet).", critical=False
         )
 
+        self.inject_warnings = self.add_par(
```

Comment:
These, and `do_warning_exception_hangup_injection_here` , are just for testing purposes, yes?

(I have failed to come up with a reason why we might ever want to use this in production.)

Reply:
Just for testing.